google_guest_agent/network/manager/netplan_linux.go

```diff
@@ -216,17 +216,32 @@ func (n *netplan) SetupEthernetInterface(ctx context.Context, config *cfg.Sectio
 
 // reloadConfigs triggers config reload to make sure ethernet/vlan configs are written
 // on disk are applied by netplan.
-func (n *netplan) reloadConfigs(ctx context.Context) error {
+func (n *netplan) reloadConfigs(ctx context.Context, interfaces map[string]bool) error {
 	logger.Infof("Reloading netplan configs...")
 
+	// Avoid restarting netplan.
+	if err := run.Quiet(ctx, "netplan", "apply"); err != nil {
+		return fmt.Errorf("error applying netplan changes: %w", err)
+	}
+
 	// Avoid restarting systemd-networkd.
 	if err := run.Quiet(ctx, "networkctl", "reload"); err != nil {
 		return fmt.Errorf("error reloading systemd-networkd network configs: %v", err)
 	}
 
-	// Avoid restarting netplan.
-	if err := run.Quiet(ctx, "netplan", "apply"); err != nil {
-		return fmt.Errorf("error applying netplan changes: %w", err)
+	// Don't attempt to reconfigure if we have no interfaces to.
+	if len(interfaces) == 0 {
+		return nil
+	}
+
+	// Force reconfigure of changed interfaces and primary interface.
+	cmdArgs := []string{"reconfigure"}
+	for k := range interfaces {
+		cmdArgs = append(cmdArgs, k)
+	}
+
+	if err := run.Quiet(ctx, "networkctl", cmdArgs...); err != nil {
```

Comment:
as per man page - `Note that this does not reload .netdev or .network corresponding to the specified interface. So, if you edit config files, it is necessary to call networkctl reload first to apply new settings.` Based on the documentation it doesn't seem like we can drop `reload` and just call `reconfigure`

Reply:
We are not dropping reload, but I agree that this will have little to no effect in fact.